src/point_one/fusion_engine/messages/configuration.h

```diff
@@ -391,6 +391,356 @@ struct alignas(4) CoarseOrientation {
+ * @ingroup config_and_ctrl_messages
+ */
+inline std::ostream& operator<<(std::ostream& stream, TransportType val) {
+  stream << to_string(val) << " (" << (int)val << ")";
+  return stream;
+}
+
+/**
+ * @brief Identifies an IO interface.
+ *
+ * (e.g., serial port 0 or TCP server 2)
+ */
+struct alignas(4) InterfaceID {
+  /** The interface's transport type. **/
+  TransportType type = TransportType::INVALID;
+  /** An identifier for the instance of this transport. */
+  uint8_t index = 0;
+  uint8_t reserved[2] = {0};
+
+  bool operator==(const InterfaceID& other) const {
+    return type == other.type && index == other.index;
+  }
+
+  bool inline operator!=(const InterfaceID& other) const {
+    return !(*this == other);
+  }
+};
+
+/**
+ * @brief @ref InterfaceID stream operator.
+ * @ingroup config_and_ctrl_messages
+ */
+inline std::ostream& operator<<(std::ostream& stream, InterfaceID val) {
+  stream << "[type=" << to_string(val.type) << ", index=" << (int)val.index
+         << "]";
+  return stream;
+}
+
+/**
+ * @brief The ways that this configuration message can be applied to the
+ *        previous list of values for that configuration type.
+ */
+enum class UpdateAction : uint8_t {
+  /**
+   * Replace the previous list of values with the set provided in
+   * this configuration.
+   */
+  REPLACE = 0
+};
+
+/**
+ * @brief Get a human-friendly string name for the specified @ref
+ *        UpdateAction.
+ * @ingroup config_and_ctrl_messages
+ *
+ * @param val The enum to get the string name for.
+ *
+ * @return The corresponding string name.
+ */
+inline const char* to_string(UpdateAction val) {
+  switch (val) {
+    case UpdateAction::REPLACE:
+      return "Replace";
+    default:
+      return "Unrecognized";
+  }
+}
+
+/**
+ * @brief @ref UpdateAction stream operator.
+ * @ingroup config_and_ctrl_messages
+ */
+inline std::ostream& operator<<(std::ostream& stream, UpdateAction val) {
+  stream << to_string(val) << " (" << (int)val << ")";
+  return stream;
+}
+
+/**
+ * @brief Configure which stream(s) will be sent to an output interface.
+ *
+ * This message is followed by `N` @ref MsgRate objects, where `N`
+ * is equal to @ref num_msgs. For example:
+ *
+ * ```
+ * {MessageHeader, SetConfigMessage, OutputStreamMsgsConfig,
+ *  MsgRate, MsgRate,  ...}
+ * ```
+ */
+struct alignas(4) OutputStreamMsgsConfig {
+  /** The stream this message configures. */
+  uint8_t stream_index = 0;
+  /**
+   * The type of action this configuration message applies to the
+   * previous state of the output stream.
+   */
+  UpdateAction update_action = UpdateAction::REPLACE;
+  /** The number of `msg_rates` entries this message contains. */
+  uint16_t num_msgs = 0;
+  /** Placeholder pointer for variable length set of messages. */
+  MsgRate msg_rates[0];
+};
+
+/**
+ * @brief Configuration for the streams associated with an output interface.
+ *
+ * This message is followed by `N` `uint8_t` stream indices, where `N`
```

Comment:
"This object is used in the payload of a @ref SetOutputInterfaceStreamsMessage message, and is followed by ..."